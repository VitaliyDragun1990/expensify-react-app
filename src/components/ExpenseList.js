import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem'
import selectExpenses from '../selectors/expenses';

export const ExpenseList = (props) => (
    <div>
        {
            props.expenses.length === 0 ? (
                <p>No expenses</p>
            ) : (
                props.expenses.map((expense) => <ExpenseListItem key={expense.id} {...expense}/>)
            )
        }
    </div>
);

// map our application state to this particular component props object
const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    };
};

// This is return High Order Component
export default connect(mapStateToProps)(ExpenseList);