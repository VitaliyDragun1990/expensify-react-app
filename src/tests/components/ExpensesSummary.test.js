import React from 'react';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should correctly render ExpensesSummary without expenses', () => {
   const wrapper = shallow(<ExpensesSummary expenseCount={0} expensesTotal={0}/>);
   expect(wrapper).toMatchSnapshot();
});

test('should correctly render ExpensesSummary with expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={expenses[0].amount}/>);
    expect(wrapper).toMatchSnapshot();
});