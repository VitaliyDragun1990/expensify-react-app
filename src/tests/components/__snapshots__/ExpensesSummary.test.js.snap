// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummary with expenses 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $1.95
  </h1>
</div>
`;

exports[`should correctly render ExpensesSummary without expenses 1`] = `
<div>
  <h1>
    Viewing 
    0
     
    expense
     totalling 
    $0.00
  </h1>
</div>
`;
